### PR TITLE
Add possibility to opt out of margin on divider

### LINKED ISSUE #805: [Enhancement] Configurable margin on kirby-divider
**Is your enhancement request related to a problem? Please describe.**
Separating two (or more) `kirby-item`s using a `kirby-divider` adds excessive margin to the bottom of the `kirby-item` before it, and excessive margin to the top of the `kirby-item` after it.

**Describe the solution you'd like**
Either completely remove `margin-top` and `margin-bottom` from `kirby-divider` or create a option that can disable the margin.

**Describe alternatives you've considered**
Using `::ng-deep` but then I felt guilty and decided to file an enhancement request instead.

**Additional context**
It looks like the browser's default styling or `hr`-elements is "bleeding through.

Example from DRB-app:
image

Dev-tools, indicating user agent stylesheet "bleeding through":
image

### PR DESCRIPTION
**Pull request by MortenGregersen**
Closes #805

### CODE REVIEW

File: src/kirby/components/divider/divider.component.scss [hunk 1]
```diff
@@ -12,4 +12,11 @@
 hr {
   border: 0;
   border-top: 1px solid var(--kirby-divider-color, get-color('medium'));
+
+  &.no-margin {
+    margin-top: 0;
```
Comment (askarby) [inline src/kirby/components/divider/divider.component.scss]: I'm pretty sure that `margin-top` and `margin-block-start` is "the same" (that same goes for `margin-bottom` and `margin-block-end`?

Furthermore, we should have an explicit size variable, setting the `margin-top|bottom` when the margin is not disabled. As it is now, the user agent stylesheet simply bleeds through, and will result in a different margin size between browsers. Consider using the kirby `size`-function.

File: src/kirby/components/divider/divider.component.ts [hunk 2]
```diff
@@ -1,8 +1,11 @@
...
 
 @Component({
   selector: 'kirby-divider',
   templateUrl: './divider.component.html',
   styleUrls: ['./divider.component.scss'],
 })
-export class DividerComponent {}
+export class DividerComponent {
+  @Input()
+  hasMargin: boolean;
```
Comment (askarby) [inline src/kirby/components/divider/divider.component.ts]: Shouldn't this be initialized to `true`, to be backwards compatible with existing usage of the component?

Comment (MortenGregersen) [reply src/kirby/components/divider/divider.component.ts]: As of @bommariusser's comment (https://github.com/kirbydesign/designsystem/issues/805#issuecomment-561622108), this should be `false` initially

Comment (MortenGregersen) [reply src/kirby/components/divider/divider.component.ts]: I have added a "BREAKING CHANGE" label

File: src/kirby/components/divider/divider.component.scss (hunk 1 above)
Comment (MortenGregersen) [reply src/kirby/components/divider/divider.component.scss]: Fixed 👍

**Pull request merged**